cpp/src/decisiontree/levelalgo/levelkernel_classifier.cuh

```diff
@@ -162,6 +162,18 @@ struct GiniDevFunctor {
     }
     return gval;
   }
+  static DI void execshared(const unsigned int* hist, float* metric,
```

Comment:
More of a general question, but would it make sense (perhaps after 0.12) to make type templates for `hist`, `metric`, and `nrows`? I've run into several instances where I've had to change things like `nrows` to be a `size_t`